src/generator/operations.ts

```diff
@@ -756,15 +756,36 @@ function createProtocolErrHandler(client: string, op: Operation, imports: Import
   // if the response doesn't define any error types return a generic error
   if (!op.exceptions) {
     imports.add('errors');
-    text += `\treturn errors.New(resp.Status)\n`;
-    text += '}\n\n';
+    imports.add('io/ioutil');
+    imports.add('fmt');
+    text += `body, err := ioutil.ReadAll(resp.Body)
+      if err != nil {
+        return fmt.Errorf("%s; failed to read response body: %w", resp.Status, err)
+      }
+      if len(body) == 0 {
+        return errors.New(resp.Status)
+      }
+      return errors.New(string(body))
+    }
+    
+    `;
     return text;
   }
   const generateUnmarshaller = function (exception: Response, prefix: string) {
     let unmarshaller = '';
     if (exception.language.go!.genericError) {
       imports.add('errors');
-      unmarshaller += `${prefix}return errors.New(resp.Status)\n`;
+      imports.add('io/ioutil');
```

Comment:
Since this has grown probably should refactor it into a single definition to avoid the duplication (can probably do the same thing as `generateUnmarshaller` above).

Reply:
Done